lib/cli/kit/system.rb

```diff
@@ -199,17 +210,30 @@ def delegate_open3(*a, sudo: raise, env: raise, method: raise, **kwargs, &block)
         # See https://github.com/Shopify/dev/pull/625 for more details.
         def resolve_path(a, env)
           # If only one argument was provided, make sure it's interpreted by a shell.
-          return ["true ; " + a[0]] if a.size == 1
+          if a.size == 1
+            if os == :windows
+              return ["break && " + a[0]]
+            else
+              return ["true ; " + a[0]]
+            end
+          end
           return a if a.first.include?('/')
 
-          paths = env.fetch('PATH', '').split(':')
-          item = paths.detect do |f|
-            command_path = "#{f}/#{a.first}"
-            File.executable?(command_path) && File.file?(command_path)
+          item = which(a.first, env)
+          a[0] = item if item
+          a
+        end
+
+        def which(cmd, env)
+          exts = os == :windows ? env.fetch('PATHEXT').split(';') : ['']
+          env.fetch('PATH', '').split(File::PATH_SEPARATOR).each do |path|
+            exts.each do |ext|
+              exe = File.join(File.expand_path(path), "#{cmd}#{ext}")
```

Comment:
Is the expand_path necessary? If it is, this could be written as `File.expand_path("#{cmd}#{ext}", path)`

Reply:
It was not necessary, I added it thinking it solved a problem but didn't try removing it afterwards. Thanks!